corehq/form_processor/models.py

```diff
@@ -39,5 +43,49 @@ def get(cls, id):
     def form_id(self):
         return self.form_uuid
 
+    def get_xml_element(self):
+        xml = self._get_xml()
+        if not xml:
+            return None
+
+        def _to_xml_element(payload):
+            if isinstance(payload, unicode):
+                payload = payload.encode('utf-8', errors='replace')
+            return etree.fromstring(payload)
+        return _to_xml_element(xml)
+
+    @property
+    def form_data(self):
+        from .utils import convert_xform_to_json
+        xml = self._get_xml()
+        return convert_xform_to_json(xml)
+
+    def _get_xml(self):
+        xform_attachment = self.xformattachmentsql_set.filter(name='form.xml').first()
+        return xform_attachment.read_content()
+
+
+class XFormAttachmentSQL(models.Model):
```

Comment:
+1 for the ability to do this, this'll probably come up time and again and a schema is super helpful for json. i'm still little wary of putting more things into the XFormInstance. with proper indexes it should be pretty quick to fetch the attachments if needed. i don't have a good sense for how often attachments are used, a few reports and restore? it'll probably make it easier to keep in the same table for partitioning though, so that could potentially be a huge simplicity gain